### PR TITLE
Add D205 pydocstyle check

### PR DESCRIPTION
**Pull request by ktrueda**
Enabled D205 Pydocstyle check

  - Fixed all pre-commit errors

Part of issue: #10742

### CODE REVIEW

**Comment by kaxil** [comment]
Static checks are failing::


```

airflow/providers/amazon/aws/sensors/step_function_execution.py:28:111: E501 line too long (118 > 110 characters)
airflow/jobs/backfill_job.py:290:111: E501 line too long (142 > 110 characters)
airflow/jobs/backfill_job.py:711:111: E501 line too long (122 > 110 characters)
airflow/jobs/backfill_job.py:768:111: E501 line too long (129 > 110 characters)
airflow/jobs/backfill_job.py:847:111: E501 line too long (147 > 110 characters)
airflow/providers/amazon/aws/example_dags/example_google_api_to_s3_transfer_advanced.py:18:111: E501 line too long (167 > 110 characters)
airflow/operators/generic_transfer.py:27:111: E501 line too long (124 > 110 characters)
airflow/providers/google/cloud/hooks/kubernetes_engine.py:89:111: E501 line too long (124 > 110 characters)
airflow/providers/exasol/hooks/exasol.py:178:111: E501 line too long (120 > 110 characters)
airflow/settings.py:127:111: E501 line too long (161 > 110 characters)
airflow/providers/google/suite/transfers/gcs_to_gdrive.py:35:111: E501 line too long (115 > 110 characters)
airflow/providers/amazon/aws/hooks/batch_client.py:239:111: E501 line too long (116 > 110 characters)
airflow/providers/amazon/aws/hooks/batch_client.py:282:111: E501 line too long (129 > 110 characters)
airflow/providers/http/sensors/http.py:29:111: E501 line too long (123 > 110 characters)
airflow/providers/amazon/aws/hooks/s3.py:47:111: E501 line too long (132 > 110 characters)
airflow/jobs/scheduler_job.py:552:111: E501 line too long (131 > 110 characters)
airflow/jobs/scheduler_job.py:939:111: E501 line too long (122 > 110 characters)
airflow/jobs/scheduler_job.py:1193:111: E501 line too long (116 > 110 characters)
airflow/jobs/scheduler_job.py:1564:111: E501 line too long (146 > 110 characters)
airflow/jobs/scheduler_job.py:1776:111: E501 line too long (130 > 110 characters)
airflow/providers/google/cloud/transfers/bigquery_to_mysql.py:31:111: E501 line too long (131 > 110 characters)
airflow/providers/amazon/aws/operators/glue.py:30:111: E501 line too long (112 > 110 characters)
airflow/configuration.py:56:111: E501 line too long (139 > 110 characters)
airflow/configuration.py:230:111: E501 line too long (122 > 110 characters)
airflow/providers/amazon/aws/example_dags/example_imap_attachment_to_s3.py:19:111: E501 line too long (144 > 110 characters)
airflow/providers/google/cloud/hooks/functions.py:61:111: E501 line too long (117 > 110 characters)
airflow/utils/log/colored_log.py:43:111: E501 line too long (140 > 110 characters)
airflow/example_dags/example_branch_python_dop_operator_3.py:20:111: E501 line too long (141 > 110 characters)
airflow/providers/google/cloud/operators/gcs.py:283:111: E501 line too long (134 > 110 characters)

```

File: airflow/__init__.py [hunk 1]
```diff
@@ -18,9 +18,9 @@
 #
 
 """
-Authentication is implemented using flask_login and different environments can
-implement their own login mechanisms by providing an `airflow_login` module
-in their PYTHONPATH. airflow_login should be based off the
+Authentication is implemented using flask_login and different environments can implement their own login mechanisms by providing an `airflow_login` module in their PYTHONPATH.
```
Comment (ashb) [inline airflow/__init__.py]: This line is too long. Please wrap at 110 characters.

File: airflow/__init__.py [hunk 2]
```diff
@@ -18,9 +18,9 @@
...
+
+airflow_login should be based off the
 `airflow.www.login`
```
Comment (ashb) [inline airflow/__init__.py]:
```suggestion
airflow_login should be based off the ``airflow.www.login``
```

File: airflow/cli/commands/task_command.py [hunk 3]
```diff
@@ -220,6 +221,8 @@ def task_run(args, dag=None):
 @cli_utils.action_logging
 def task_failed_deps(args):
     """
+    task_failed_deps.
```
Comment (ashb) [inline airflow/cli/commands/task_command.py]:
```suggestion
    List which dependencies are failing (stopping execution) for a given TaskInstance.
```

File: airflow/cli/commands/webserver_command.py [hunk 4]
```diff
@@ -46,6 +46,8 @@
 
 class GunicornMonitor(LoggingMixin):
     """
+    GunicornMonitor
+
```
Comment (ashb) [inline airflow/cli/commands/webserver_command.py:51]:
```suggestion
```
The existing first paragraph _was_ a good summary.

File: airflow/configuration.py [hunk 5]
```diff
@@ -53,9 +53,8 @@
 
 def expand_env_var(env_var):
     """
-    Expands (potentially nested) env vars by repeatedly applying
-    `expandvars` and `expanduser` until interpolation stops having
-    any effect.
+    Expands (potentially nested) env vars by repeatedly applying `expandvars` and `expanduser` until interpolation stops having any effect.
+
```
Comment (ashb) [inline airflow/configuration.py]: The previous version was fine as it was.

File: airflow/configuration.py [hunk 6]
```diff
@@ -228,8 +227,8 @@ def _validate(self):
 
     def _validate_config_dependencies(self):
         """
-        Validate that config values aren't invalid given other config values
-        or system-level limitations and requirements.
+        Validate that config values aren't invalid given other config values or system-level limitations and requirements.
+
```
Comment (ashb) [inline airflow/configuration.py]: Previous form was fine/correct.

File: airflow/configuration.py [hunk 7]
```diff
@@ -986,6 +985,8 @@ def get_custom_secret_backend() -> Optional[BaseSecretsBackend]:
 
 def initialize_secrets_backends() -> List[BaseSecretsBackend]:
     """
+    initialize_secrets_backends
```
Comment (ashb) [inline airflow/configuration.py]:
```suggestion
    Initialize configured and built-in secrets backends 
```

File: airflow/contrib/operators/file_to_gcs.py [hunk 8]
```diff
@@ -31,7 +31,8 @@
 
 class FileToGoogleCloudStorageOperator(LocalFilesystemToGCSOperator):
     """
-    This class is deprecated.
+    This class is deprecated
```
Comment (ashb) [inline airflow/contrib/operators/file_to_gcs.py]:
```suggestion
    This class is deprecated.
```

File: airflow/dag/base_dag.py [hunk 9]
```diff
@@ -30,6 +30,8 @@ class BaseDag(metaclass=ABCMeta):
     @abstractmethod
     def dag_id(self):
         """
+        dag_id.
+
         :return: the DAG ID
```
Comment (ashb) [inline airflow/dag/base_dag.py]:
```suggestion
        The DAG ID

```

File: airflow/dag/base_dag.py [hunk 10]
```diff
@@ -39,6 +41,8 @@ def dag_id(self):
     @abstractmethod
     def task_ids(self):
         """
+        task_ids.
+
         :return: A list of task IDs that are in this DAG
```
Comment (ashb) [inline airflow/dag/base_dag.py]:
```suggestion
        A list of task IDs that are in this DAG

```

File: airflow/dag/base_dag.py [hunk 11]
```diff
@@ -48,6 +52,8 @@ def task_ids(self):
     @abstractmethod
     def full_filepath(self):
         """
+        full_filepath.
+
         :return: The absolute path to the file that contains this DAG's definition
```
Comment (ashb) [inline airflow/dag/base_dag.py]:
```suggestion
        The absolute path to the file that contains this DAG's definition

```

File: airflow/dag/base_dag.py [hunk 12]
```diff
@@ -57,6 +63,8 @@ def full_filepath(self):
     @abstractmethod
     def concurrency(self):
         """
+        concurrency.
+
         :return: maximum number of tasks that can run simultaneously from this DAG
```
Comment (ashb) [inline airflow/dag/base_dag.py]:
```suggestion
        Maximum number of tasks that can run simultaneously from this DAG

```

File: airflow/dag/base_dag.py [hunk 13]
```diff
@@ -65,6 +73,8 @@ def concurrency(self):
     @abstractmethod
     def pickle_id(self):
         """
+        pickle_id.
+
         :return: The pickle ID for this DAG, if it has one. Otherwise None.
```
Comment (ashb) [inline airflow/dag/base_dag.py]:
```suggestion
        The pickle ID for this DAG, if it has one, otherwise None.

```

File: airflow/dag/base_dag.py [hunk 14]
```diff
@@ -80,6 +90,8 @@ class BaseDagBag:
     @abstractmethod
     def dag_ids(self):
         """
+        dag_ids.
+
         :return: a list of DAG IDs in this bag
```
Comment (ashb) [inline airflow/dag/base_dag.py]:
```suggestion
        A list of DAG IDs in this bag

```

File: airflow/dag/base_dag.py [hunk 15]
```diff
@@ -88,6 +100,8 @@ def dag_ids(self):
     @abstractmethod
     def get_dag(self, dag_id):
         """
+        get_dag.
+
         :return: whether the task exists in this bag
```
Comment (ashb) [inline airflow/dag/base_dag.py]:
```suggestion
        Get a DAG from this bag, if it exists, or else return None.

```

File: airflow/example_dags/example_branch_python_dop_operator_3.py [hunk 16]
```diff
@@ -17,8 +17,7 @@
 # under the License.
 
 """
-Example DAG demonstrating the usage of BranchPythonOperator with depends_on_past=True, where tasks may be run
-or skipped on alternating runs.
+Example DAG demonstrating the usage of BranchPythonOperator with depends_on_past=True, where tasks may be run or skipped on alternating runs.
```
Comment (ashb) [inline airflow/example_dags/example_branch_python_dop_operator_3.py]: Revert this -- previous one was fine.

File: airflow/example_dags/example_external_task_marker_dag.py [hunk 17]
```diff
@@ -17,8 +17,7 @@
 # under the License.
 
 """
-Example DAG demonstrating setting up inter-DAG dependencies using ExternalTaskSensor and
-ExternalTaskMarker
+Example DAG demonstrating setting up inter-DAG dependencies using ExternalTaskSensor and ExternalTaskMarker
```
Comment (ashb) [inline airflow/example_dags/example_external_task_marker_dag.py:20]:
```suggestion
Example DAG demonstrating setting up inter-DAG dependencies using ExternalTaskSensor and ExternalTaskMarker.
```

**Review by ashb — CHANGES_REQUESTED** [review]
I've left some review, but you have made many of the lines too long, and (needlessly?) changed the line wrapping -- please don't do that unless there is a reason.

For example, this sort of change is not needed:

```diff
@@ -1856,8 +1861,7 @@ def get_default_view(self):
    @provide_session
    def deactivate_unknown_dags(active_dag_ids, session=None):
        """
-        Given a list of known DAGs, deactivate any other DAGs that are
-        marked as active in the ORM
+        Given a list of known DAGs, deactivate any other DAGs that are marked as active in the ORM
        :param active_dag_ids: list of DAG IDs that are active
        :type active_dag_ids: list[unicode]
```

File: airflow/__init__.py (hunk 2 above)
Comment (ktrueda) [reply airflow/__init__.py]: Thank you.

File: airflow/cli/commands/task_command.py (hunk 3 above)
Comment (ktrueda) [reply airflow/cli/commands/task_command.py]: Thank you

File: airflow/cli/commands/webserver_command.py (hunk 4 above)
Comment (ktrueda) [reply airflow/cli/commands/webserver_command.py:51]: The existing first paragraph was good. But it's long as a first line summary .
Obeying D205 and E501 summary line needs to obey two rules.
- only 1 line (not 2+ line)
- length < 110

**Comment by mik-laj** [comment]
Static checks are sad.
```
airflow/kubernetes/kube_client.py:69 in private function `_enable_tcp_keepalive`:
        D205: 1 blank line required between summary line and description (found 0)
airflow/providers/discord/operators/discord_webhook.py:28 in public class `DiscordWebhookOperator`:
        D205: 1 blank line required between summary line and description (found 0)
airflow/providers/google/cloud/operators/bigquery_dts.py:199 in public class `BigQueryDataTransferServiceStartTransferRunsOperator`:
        D205: 1 blank line required between summary line and description (found 0)
airflow/providers/microsoft/azure/operators/azure_cosmos.py:25 in public class `AzureCosmosInsertDocumentOperator`:
        D205: 1 blank line required between summary line and description (found 0)
```

**Comment by ktrueda** [comment]
`CI Build / Build docs (pull_request) Failing after 10m — Build docs` said

```
==================== Error   1 ====================
 WARNING: Definition list ends without a blank line; unexpected unindent.

File path: /opt/airflow/docs/_api/airflow/providers/slack/operators/slack/index.rst (71)

  67 |            dag=dag,
  68 |            token="XXX",
  69 |            text="hello there!",
  70 |            channel="#random",
  71 |        )
  72 | 
  73 |    :param channel: channel in which to post message on slack name (#general) or
  74 |        ID (C12318391). (templated)
  75 |    :type channel: str
  76 |    :param username: Username that airflow will be posting to Slack as. (templated)
==================== Error   2 ====================
 WARNING: Definition list ends without a blank line; unexpected unindent.

File path: /opt/airflow/docs/_api/airflow/providers/slack/operators/slack/index.rst (123)

 119 |            initial_comment="Hello World!",
 120 |            filename="hello_world.csv",
 121 |            filetype="csv",
 122 |            content="hello,world,csv,file",
 123 |        )
 124 | 
 125 |    :param channel: channel in which to sent file on slack name (templated)
 126 |    :type channel: str
 127 |    :param initial_comment: message to send to slack. (templated)
 128 |    :type initial_comment: str
==================================================
```

`CI Build / Spell check docs (pull_request)` said

```
/opt/airflow/docs/_api/airflow/providers/slack/operators/slack/index.rst:71:Definition list ends without a blank line; unexpected unindent.
==================== Error   1 ====================
Sphinx spellcheck returned non-zero exit status: 2.

==================================================
```

I didn't modify these files. How can I pass these CI.

And `CI Build / MySQL8, Py3.6 (pull_request)` failed. I have no idea about this test.

**Comment by potiuk** [comment]
I believe it was a merge of a change that failed the static checks -  it is already fixed by https://github.com/apache/airflow/commit/95be3eec42cd5ae0ef8c6402f7d1cd87e5d93848 and  you need to rebase to latest master to fix it (that's what I did with my PRs)

**Comment by ktrueda** [comment]
@potiuk 
Thank you for your advice!!!
I rebased it and pushed it.

🙏

**Comment by ktrueda** [comment]
@potiuk 
I tried rebase, but it doesn't work.

File: airflow/providers/slack/operators/slack.py [hunk 18]
```diff
@@ -89,9 +92,9 @@ def execute(self, **kwargs):  # noqa: D403
 
 class SlackAPIPostOperator(SlackAPIOperator):
     """
-    Posts messages to a slack channel
-    Examples:
+    Posts messages to a slack channel.
 
+    Examples:
     .. code-block:: python
```
Comment (kaxil) [inline airflow/providers/slack/operators/slack.py]: Tests is failing because of this, a space is needed here I think

```suggestion
    Examples:

    .. code-block:: python
```

File: airflow/providers/slack/operators/slack.py [hunk 19]
```diff
@@ -159,8 +162,8 @@ def construct_api_call_params(self) -> Any:
 class SlackAPIFileOperator(SlackAPIOperator):
     """
     Send a file to a slack channel
-    Examples:
 
+    Examples:
     .. code-block:: python
```
Comment (kaxil) [inline airflow/providers/slack/operators/slack.py]:
```suggestion
    Examples:

    .. code-block:: python
```

**Comment by ktrueda** [comment]
@kaxil 
Thank you for your suggestion very much!!

🙏

File: airflow/providers/slack/operators/slack.py (hunk 18 above)
Comment (ktrueda) [reply airflow/providers/slack/operators/slack.py]: @kaxil 
I merged this suggestion and run pre-commit.

I found that this suggestion is illegal about D412(No blank lines allowed between a section header and its content)

Comment (ktrueda) [reply airflow/providers/slack/operators/slack.py]: So I think that 
if I obey D412, I can't pass the static check,
if i pass the static check, I can't obey D412 (not confirmed.)

Comment (kaxil) [reply airflow/providers/slack/operators/slack.py]: Don't worry, I will check back later tonight or tomorrow and we will figure out a solution, you have done an amazing job with this PR, thanks.


Btw there are some conflicts, can you rebase on the lastest master please

Comment (ktrueda) [reply airflow/providers/slack/operators/slack.py]: I see. Thank you.

I've rebased and pushed.

**Comment by ktrueda** [comment]
@kaxil 
https://github.com/apache/airflow/pull/11627#discussion_r511490113
If you have any solution, please let me know it.

**Comment by kaxil** [comment]
@ktrueda I have rebased on master, resolved the merge conflicts, pushed a fix. Let's see what the CI says.

**Comment by mik-laj** [comment]
CI is sad. I working on fix. 
```
black.............................................................................................................................Failed
- hook id: black
- files were modified by this hook

reformatted /home/runner/work/airflow/airflow/airflow/providers/slack/operators/slack.py
All done! ✨ 🍰 ✨
1 file reformatted, 1558 files left unchanged.
```
```
All changes made by hooks:
diff --git a/airflow/providers/slack/operators/slack.py b/airflow/providers/slack/operators/slack.py
index 18bc86f..29b53cf 100644
--- a/airflow/providers/slack/operators/slack.py
+++ b/airflow/providers/slack/operators/slack.py
@@ -90,7 +90,7 @@ class SlackAPIOperator(BaseOperator):
         slack.call(self.method, json=self.api_params)
 
 
-class SlackAPIPostOperator(SlackAPIOperator):   # noqa: D412
+class SlackAPIPostOperator(SlackAPIOperator):  # noqa: D412
     """
     Posts messages to a slack channel.
 
@@ -160,7 +160,7 @@ class SlackAPIPostOperator(SlackAPIOperator):   # noqa: D412
         }
 
 
-class SlackAPIFileOperator(SlackAPIOperator):   # noqa: D412
+class SlackAPIFileOperator(SlackAPIOperator):  # noqa: D412
     """
     Send a file to a slack channel
 
Error: Process completed with exit code 1.
```

**Comment by mik-laj** [comment]
Spell checks are sad now. :-/
```
_api/airflow/models/dag/index.rst:1109: (parametrize)  Accepts kwargs for operator kwarg. Can be used to parametrize DAGs.
```

File: airflow/models/dag.py [hunk 20]
```diff
@@ -388,6 +390,8 @@ def __exit__(self, _type, _value, _tb):
     @staticmethod
     def _upgrade_outdated_dag_access_control(access_control=None):
         """
+        _upgrade_outdated_dag_access_control
```
Comment (mik-laj) [inline airflow/models/dag.py]: It looks unreleated

File: airflow/models/dag.py [hunk 21]
```diff
@@ -525,6 +531,8 @@ def next_dagrun_info(
 
     def next_dagrun_after_date(self, date_last_automated_dagrun: Optional[pendulum.DateTime]):
         """
+        next_dagrun_after_date
```
Comment (mik-laj) [inline airflow/models/dag.py]: It looks unrelated.

File: airflow/models/dag.py [hunk 22]
```diff
@@ -810,6 +817,8 @@ def normalized_schedule_interval(self) -> Optional[ScheduleInterval]:
     @provide_session
     def handle_callback(self, dagrun, success=True, reason=None, session=None):
         """
+        handle_callback
```
Comment (mik-laj) [inline airflow/models/dag.py]: It looks unrelated.

**Comment by kaxil** [comment]
> parametrize

Fixed spelling and pushed

File: airflow/models/dag.py (hunk 20 above)
Comment (kaxil) [reply airflow/models/dag.py]: We should not be adding a dummy Summary heading for docstring @ktrueda

**Comment by ktrueda** [comment]
@kaxil @mik-laj 
Thank you for fixing very much.
I'll fix dummy summary heading.

**Comment by mik-laj** [comment]
@ktrueda  CI is sad. Can you do a rebase?

**Comment by ktrueda** [comment]
> @ktrueda CI is sad. Can you do a rebase?

Thank you.
I've rebased.

File: airflow/__init__.py [hunk 23]
```diff
@@ -18,6 +18,8 @@
 #
 
 """
+__init__.py
```
Comment (kaxil) [inline airflow/__init__.py]: this is a Dummy header

File: airflow/api/common/experimental/mark_tasks.py [hunk 24]
```diff
@@ -218,6 +221,8 @@ def verify_dagruns(dag_runs, commit, state, session, current_task):
 
 def verify_dag_run_integrity(dag, dates):
     """
+    verify_dag_run_integrity
```
Comment (kaxil) [inline airflow/api/common/experimental/mark_tasks.py]: this is a Dummy header

File: airflow/cli/commands/webserver_command.py [hunk 25]
```diff
@@ -105,6 +107,8 @@ def __init__(
 
     def _generate_plugin_state(self) -> Dict[str, float]:
         """
+        _generate_plugin_state
```
Comment (kaxil) [inline airflow/cli/commands/webserver_command.py]: this is a Dummy header

File: airflow/configuration.py [hunk 26]
```diff
@@ -52,6 +52,8 @@
 
 def expand_env_var(env_var):
     """
+    expand_env_var
```
Comment (kaxil) [inline airflow/configuration.py:56]: this is a Dummy header

File: airflow/configuration.py [hunk 27]
```diff
@@ -225,6 +227,8 @@ def _validate(self):
 
     def _validate_config_dependencies(self):
         """
+        _validate_config_dependencies
```
Comment (kaxil) [inline airflow/configuration.py:231]: There are lot of dummy headers instead of actual headers that needs to be fixed

**Comment by ktrueda** [comment]
@kaxil 
Thank you for your comment.
I'm currently working on it. I need some times since there are a lot of dummies.

**Comment by kaxil** [comment]
Can you please rebase your PR on latest Master since we have applied [Black](https://github.com/apache/airflow/commit/4e8f9cc8d02b29c325b8a5a76b4837671bdf5f68) and [PyUpgrade](https://github.com/apache/airflow/commit/8c42cf1b00c90f0d7f11b8a3a455381de8e003c5) on Master.

It will help if your squash your commits into single commit first so that there are less conflicts.

**Comment by ktrueda** [comment]
@kaxil 
I rebased and wrote summary line instead of dummy. 
I did my best but some text is redundant and some is same with function/class name. If you have better idea, please suggest.
I'm sorry for my lack of skill and understanding the project.

**Comment by ktrueda** [comment]
@kaxil 
If my code is bad, please suggest. 
Or I'll close this PR.

**Comment by kaxil** [comment]
> @kaxil
> If my code is bad, please suggest.
> Or I'll close this PR.

Thanks @ktrueda , I am on leave today, will check back once I am back
